nflow-engine/src/main/java/io/nflow/engine/service/MaintenanceService.java

```diff
@@ -76,6 +83,17 @@ public MaintenanceResults cleanupWorkflows(MaintenanceConfiguration configuratio
     return builder.build();
   }
 
+  private void validateConfiguration(MaintenanceConfiguration configuration) {
+    Stream.of(configuration.archiveWorkflows, configuration.deleteArchivedWorkflows, configuration.deleteWorkflows) //
+        .filter(Objects::nonNull) //
+        .flatMap(configItem -> configItem.workflowTypes.stream()) //
+        .forEach(workflowType -> {
```

Comment:
You could do more stream/optional stuff with the following, but I'm not sure if it looks nicer or not
```java
            .filter(workflowType -> workflowDefinitionService.getWorkflowDefinition(workflowType) == null)
            .findAny()
            .ifPresent(workflowType -> { throw new IllegalArgumentException("Workflow type " + workflowType + " is not valid"); });
```

Reply:
Yes it looks nicer.